src/SFA.DAS.Commitments.Application.UnitTests/Queries/GetApprenticeship/WhenGettingApprenticeship.cs

```diff
@@ -0,0 +1,90 @@
+﻿using NUnit.Framework;
+using System.Threading.Tasks;
+using SFA.DAS.Commitments.Domain.Data;
+using Moq;
+using SFA.DAS.Commitments.Domain;
+using FluentAssertions;
+using System;
+using SFA.DAS.Commitments.Application.Exceptions;
+using Ploeh.AutoFixture;
+using SFA.DAS.Commitments.Application.Queries.GetApprenticeship;
+
+namespace SFA.DAS.Commitments.Application.UnitTests.Queries.GetApprenticeship
+{
+    [TestFixture]
+    public sealed class WhenGettingApprenticeship
+    {
+        private Mock<ICommitmentRepository> _mockCommitmentRespository;
+        private GetApprenticeshipQueryHandler _handler;
+        private GetApprenticeshipRequest _exampleValidRequest;
+        private Commitment _fakeRepositoryCommitment;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockCommitmentRespository = new Mock<ICommitmentRepository>();
+            _handler = new GetApprenticeshipQueryHandler(_mockCommitmentRespository.Object, new GetApprenticeshipValidator());
+
+            Fixture dataFixture = new Fixture();
+            _fakeRepositoryCommitment = dataFixture.Build<Commitment>().Create();
+            
+            _exampleValidRequest = new GetApprenticeshipRequest { AccountId = _fakeRepositoryCommitment.EmployerAccountId, CommitmentId = _fakeRepositoryCommitment.Id, ApprenticeshipId = _fakeRepositoryCommitment.Apprenticeships[0].Id };
+        }
+
+        [Test]
+        public async Task ThenTheCommitmentRepositoryIsCalled()
+        {
+            await _handler.Handle(_exampleValidRequest);
+
+            _mockCommitmentRespository.Verify(x => x.GetById(It.IsAny<long>()), Times.Once);
+        }
+
+        [Test]
+        public async Task ThenShouldReturnAnApprenticeshipInResponse()
+        {
+            _mockCommitmentRespository.Setup(x => x.GetById(It.IsAny<long>())).ReturnsAsync(_fakeRepositoryCommitment);
+
+            var response = await _handler.Handle(_exampleValidRequest);
+
+            response.Data.Id.Should().Be(_fakeRepositoryCommitment.Apprenticeships[0].Id);
+            response.Data.ApprenticeName.Should().Be(_fakeRepositoryCommitment.Apprenticeships[0].ApprenticeName);
+        }
+
+        [Test]
+        public void ThenIfCommitmentIdIsZeroItThrowsAnInvalidRequestException()
+        {
+            Func<Task> act = async () => await _handler.Handle(new GetApprenticeshipRequest { CommitmentId = 0 });
```

Comment:
Have you seen this style?
var exception = Assert.ThrowsAsync<InvalidRequestException>(() => _handler.Handle(command));